test/standalone/Vegetation/test_pmodel.jl

```diff
@@ -216,3 +223,251 @@ end
+    start_date = DateTime(2005)
+
+    # Dummy radiative forcing
+    zenith_angle =
+        (t, s) -> default_zenith_angle(
+            t,
+            s;
+            insol_params = earth_param_set.insol_params,
+            latitude = lat,
+            longitude = long,
+        )
+    shortwave_radiation = (t; latitude=lat, longitude=long, insol_params=earth_param_set.insol_params) -> 1000
+    longwave_radiation = t -> 200
+    radiation = PrescribedRadiativeFluxes(
+        FT,
+        TimeVaryingInput(shortwave_radiation),
+        TimeVaryingInput(longwave_radiation),
+        start_date;
+        θs = zenith_angle,
+        earth_param_set = earth_param_set,
+    )
+
+    # Dummy atmospheric forcing 
+    u_atmos = t -> 10 #m.s-1
+    liquid_precip = (t) -> 0 # m
+    snow_precip = (t) -> 0 # m
+    T_atmos = t -> 290 # Kelvin
+    q_atmos = t -> 0.001 # kg/kg
+    P_atmos = t -> 1e5 # Pa
+    h_atmos = h_int # m
+    c_atmos = (t) -> 4.11e-4 # mol/mol
+    atmos = PrescribedAtmosphere(
+        TimeVaryingInput(liquid_precip),
+        TimeVaryingInput(snow_precip),
+        TimeVaryingInput(T_atmos),
+        TimeVaryingInput(u_atmos),
+        TimeVaryingInput(q_atmos),
+        TimeVaryingInput(P_atmos),
+        start_date,
+        h_atmos,
+        earth_param_set;
+        c_co2 = TimeVaryingInput(c_atmos),
+    )
+
+    canopy = ClimaLand.Canopy.CanopyModel{FT}(;
+        parameters = shared_params,
+        domain = canopy_domain,
+        autotrophic_respiration = ar_model,
+        radiative_transfer = rt_model,
+        photosynthesis = photosynthesis_model,
+        conductance = stomatal_model,
+        hydraulics = plant_hydraulics,
+        boundary_conditions = Canopy.AtmosDrivenCanopyBC(
+            atmos,
+            radiation,
+            soil_driver,
+        ),
+    )
+
+    @test_nowarn pmodel_callback = make_PModel_callback(FT, start_date, dt, canopy)
+end
+
+
+@testset "Test helper functions update_optimal_EMA and update_intermediate_vars correctness" begin
+    rtol = 1e-5
+    atol = 1e-6
+
+    for FT in (Float32, Float64)
+        parameters = ClimaLand.Canopy.PModelParameters(
+            cstar = FT(0.41),
+            β = FT(146),
+            ϕc = FT(0.087),
+            ϕ0 = FT(NaN),
+            ϕa0 = FT(0.352),
+            ϕa1 = FT(0.022),
+            ϕa2 = FT(-0.00034),
+            α = FT(0),
+            sc = FT(2e-6),
+            pc = FT(-2e6),
+        )
+
+        constants = PModelConstants{FT}()
+
+        T_canopy = FT(281.25)
+        I_abs = FT(0.0013948839623481035)
+        ca = FT(0.00039482)
+        P_air = FT(99230.0)
+        VPD = FT(756.2)
+        βm = FT(1.0)
+
+        outputs_full = compute_full_pmodel_outputs(
+            parameters,
+            constants,
+            T_canopy,
+            P_air,
+            VPD,
+            ca,
+            βm,
+            I_abs,
+        )
+
+        @testset "Test update_optimal_EMA optimality computation for $FT" begin
```

Comment:
These don't need to be in a nested testset - they can just be in the outer testset (same for the other testset below)